WordPress/Classes/ViewRelated/Blog/TimezoneSelectorViewModel.swift

```diff
@@ -0,0 +1,203 @@
+import Foundation
+import CoreData
+
+/// Model for TimezoneSelectorViewController's ImmuTablePresenter
+enum TimezoneSelectorViewModel {
+    case loading
+    /**
+     - parameters
+        - first param: - all TimezoneInfo objects from DB
+        - second param: - usersCurrentTimeZoneString passed from TimezoneSelectorVC
+        - third param: - usersCurrentManualGMTOffset passed from TimezoneSelectorVC
+        - fourth param: - action block to be executed when user clicks on a cell
+    */
+    case ready([TimezoneInfo], String?, NSNumber?, ImmuTableAction)
+    case error(String)
+
+    /// the first param is a list of all continents, used as section headers
+    /// the second param is a mapping from continent to timezones in that continent
+    typealias ContinentsAndTimezones = ([String], [String: [(String, String)]])
+
+    /// constant used in the functions below
+    var MANUAL_OFFSET: String { return "Manual Offsets" }
+    var noResultsViewModel: WPNoResultsView.Model? {
+        switch self {
+        case .loading:
+            return WPNoResultsView.Model(
+                title: NSLocalizedString("Loading Timezones...",
+                                         comment: "Text displayed while loading the activity feed for a site")
+            )
+        case .ready:
+            return nil
+        case .error:
+            let appDelegate = WordPressAppDelegate.sharedInstance()
+            if (appDelegate?.connectionAvailable)! {
+                return WPNoResultsView.Model(
+                    title: NSLocalizedString("Oops", comment: ""),
+                    message: NSLocalizedString("There was an error loading timezones",
+                                               comment: "Text displayed when there is a failure loading the timezone list"),
+                    buttonTitle: NSLocalizedString("Contact support",
+                                                   comment: "Button label for contacting support")
+                )
+            } else {
+                return WPNoResultsView.Model(
+                    title: NSLocalizedString("No connection", comment: ""),
+                    message: NSLocalizedString("An active internet connection is required to view timezones", comment: "")
+                )
+            }
+        }
+    }
+
+    func tableViewModel() -> ImmuTable {
+        switch self {
+        case .loading, .error:
+            return .Empty
+        case .ready(let timezoneInfoArray, let usersTimezoneString, let usersManualOffset, let action):
+            let (continents, continentToTimezones) = self.setupVariables(with: timezoneInfoArray)
+            let indexPathToHighlight = self.getIndexPathToHighlight(timezoneString: usersTimezoneString, manualOffset: usersManualOffset, data: (continents, continentToTimezones), allTimezones: timezoneInfoArray)
```

Comment:
I think this code is more complicated than it needs to be because it tries to use an `indexPath` to figure out the selected time zone (*). This would be much simpler if it just iterated through the sections/rows checking if each one should be selected or not.

(*) Side note: I understand UITableView uses `highlighted` terminology and `selected` for something else, but this is a `*Selector` component and I think it makes more sense to use "selected" for the currently selected time zone.